src/code.cloudfoundry.org/gorouter/registry/registry.go

```diff
@@ -88,32 +88,42 @@ func (r *RouteRegistry) Register(uri route.Uri, endpoint *route.Endpoint) {
 		return
 	}
 
-	endpointAdded := r.register(uri, endpoint)
+	poolPutResult, routeAdded := r.register(uri, endpoint)
 
-	r.reporter.CaptureRegistryMessage(endpoint, endpointAdded.String())
+	r.reporter.CaptureRegistryMessage(endpoint, poolPutResult.String())
 
-	if endpointAdded == route.ADDED && !endpoint.UpdatedAt.IsZero() {
+	if poolPutResult == route.EndpointAdded && !endpoint.UpdatedAt.IsZero() {
 		r.reporter.CaptureRouteRegistrationLatency(time.Since(endpoint.UpdatedAt))
 	}
 
-	switch endpointAdded {
-	case route.ADDED:
+	if routeAdded {
+		r.logger.Info("route-registered", slog.Any("uri", uri))
+		// for backward compatibility:
+		r.logger.Debug("uri-added", slog.Any("uri", uri))
+		r.reporter.CaptureRouteAdded()
+	}
+
+	switch poolPutResult {
+	case route.EndpointAdded:
 		if r.logger.Enabled(context.Background(), slog.LevelInfo) {
 			r.logger.Info("endpoint-registered", buildSlogAttrs(uri, endpoint)...)
 		}
-	case route.UPDATED:
-		if r.logger.Enabled(context.Background(), slog.LevelDebug) {
-			r.logger.Debug("endpoint-registered", buildSlogAttrs(uri, endpoint)...)
+	case route.EndpointUpdated:
+		if r.logger.Enabled(context.Background(), slog.LevelInfo) {
+			r.logger.Info("endpoint-registered", buildSlogAttrs(uri, endpoint)...)
```

Comment:
Do we want to update the log message as well? To distinguish between updated and registered?

Reply:
We decided to adjust the log messages as little as possible to not make this a breaking change.